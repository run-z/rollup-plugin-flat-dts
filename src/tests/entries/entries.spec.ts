import { describe, expect, it } from '@jest/globals';
import { testDts } from '../test-dts';

describe('With multiple entries', () => {
  it('emits multiple definition files when entry file name specified', async () => {
    const { files } = await testDts('entries', {
      entries: {
        entry1: { file: 'entry1.d.ts' },
        entry2: { file: 'entry2.d.ts' },
      },
    });

    expect(files).toHaveLength(3);

    const [{ content: file1 }, { content: file2 }, { content: file3 }] = files;

    expect(file1).toMatchSnapshot('file1');
    expect(file2).toMatchSnapshot('file2');
    expect(file3).toMatchSnapshot('file3');
  });
  it('emits one definition file when entry file name omitted', async () => {
    const { files } = await testDts('entries', {
      entries: {
        entry1: {},
        entry2: {},
      },
    });

    expect(files).toHaveLength(1);

    const [{ content }] = files;

    expect(content).toMatchSnapshot();
  });
  it('renames the entry when its final name specified', async () => {
    const { files } = await testDts('entries', {
      entries: {
        entry1: { as: 'test-entry1' },
        entry2: { as: 'test-entry2' },
      },
    });

    expect(files).toHaveLength(1);

    const [{ content }] = files;

    expect(content).toMatchSnapshot();
  });
  it('merges multiple entries with the same file name', async () => {
    const { files } = await testDts('entries', {
      entries: {
        entry1: { file: 'entries.d.ts' },
        entry2: { file: 'entries.d.ts' },
      },
    });

    expect(files).toHaveLength(2);

    const [{ content: file1 }, { content: file2 }] = files;

    expect(file1).toMatchSnapshot('file1');
    expect(file2).toMatchSnapshot('file2');
  });
});
