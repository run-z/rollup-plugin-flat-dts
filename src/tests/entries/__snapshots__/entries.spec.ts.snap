// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With multiple entries emits multiple definition files when entry file name specified: file1 1`] = `
"declare module \\"rollup-plugin-flat-dts/entry1\\" {
    export interface Entry1 {
        readonly name: 'entry1';
    }
}
"
`;

exports[`With multiple entries emits multiple definition files when entry file name specified: file2 1`] = `
"declare module \\"rollup-plugin-flat-dts/entry2\\" {
    export interface Entry2 {
        readonly name: 'entry2';
    }
}
"
`;

exports[`With multiple entries emits multiple definition files when entry file name specified: file3 1`] = `
"/// <reference path=\\"entry1.d.ts\\" />
/// <reference path=\\"entry2.d.ts\\" />
declare module \\"rollup-plugin-flat-dts\\" {
    export * from \\"rollup-plugin-flat-dts/entry1\\";
    export * from \\"rollup-plugin-flat-dts/entry2\\";
    export interface Root {
        readonly name: 'root';
    }
}
"
`;

exports[`With multiple entries emits one definition file when entry file name omitted 1`] = `
"declare module \\"rollup-plugin-flat-dts/entry1\\" {
    export interface Entry1 {
        readonly name: 'entry1';
    }
}

declare module \\"rollup-plugin-flat-dts/entry2\\" {
    export interface Entry2 {
        readonly name: 'entry2';
    }
}

declare module \\"rollup-plugin-flat-dts\\" {
    export * from \\"rollup-plugin-flat-dts/entry1\\";
    export * from \\"rollup-plugin-flat-dts/entry2\\";
    export interface Root {
        readonly name: 'root';
    }
}
"
`;

exports[`With multiple entries merges multiple entries with the same file name: file1 1`] = `
"declare module \\"rollup-plugin-flat-dts/entry1\\" {
    export interface Entry1 {
        readonly name: 'entry1';
    }
}

declare module \\"rollup-plugin-flat-dts/entry2\\" {
    export interface Entry2 {
        readonly name: 'entry2';
    }
}
"
`;

exports[`With multiple entries merges multiple entries with the same file name: file2 1`] = `
"/// <reference path=\\"entries.d.ts\\" />
/// <reference path=\\"entries.d.ts\\" />
declare module \\"rollup-plugin-flat-dts\\" {
    export * from \\"rollup-plugin-flat-dts/entry1\\";
    export * from \\"rollup-plugin-flat-dts/entry2\\";
    export interface Root {
        readonly name: 'root';
    }
}
"
`;

exports[`With multiple entries renames the entry when its final name specified 1`] = `
"declare module \\"rollup-plugin-flat-dts/test-entry1\\" {
    export interface Entry1 {
        readonly name: 'entry1';
    }
}

declare module \\"rollup-plugin-flat-dts/test-entry2\\" {
    export interface Entry2 {
        readonly name: 'entry2';
    }
}

declare module \\"rollup-plugin-flat-dts\\" {
    export * from \\"rollup-plugin-flat-dts/test-entry1\\";
    export * from \\"rollup-plugin-flat-dts/test-entry2\\";
    export interface Root {
        readonly name: 'root';
    }
}
"
`;
